### PR TITLE
Feature: Multi network support

### PR DESCRIPTION
**Pull request by chumaknadya**
- [x] - write documentation

### CODE REVIEW

File: db/migrate/20210601111215_multi_network_support.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    end
+
+    # Update all coin beneficiaries with blockchain_key
+    Beneficiary.find_each(batch_size: 100) do |beneficiary|
+      beneficiary.update(blockchain_key: beneficiary.currency.blockchain_key)
+    end
+
+    # Update all pending coin deposits with blockchain_key by wallet
+    Deposit.processing.find_each(batch_size: 100) do |deposit|
+      deposit.update(blockchain_key: deposit.currency.blockchain_key)
```
Comment (calj) [inline db/migrate/20210601111215_multi_network_support.rb]: You need to update all deposits, withdrawals and transactions, not only processing.
In order to have the network in history.

File: app/models/blockchain_currency.rb [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        # Link parent currency with wallet
+        CurrencyWallet.create(currency_id: currency_id, wallet_id: wallet.id)
+      end
+    end
+  end
+
+  def update_fees
+    market = Market.find_by(base_unit: currency.id, quote_unit: Peatio::App.config.platform_currency)
+    ticker = Trade.market_ticker_from_influx(market.symbol) if market.present?
+    price = ticker.present? ? ticker[:vwap].to_d : currency.price
```
Comment (calj) [inline app/models/blockchain_currency.rb]: we need to take this opportunity to update the price field in currency.
better to have a specific method to update the price so we can run it even if the auto fee update is not activated

File: config/templates/seed/blockchains.yml.erb [hunk 3]
```diff
@@ -22,6 +23,7 @@
     address:         https://rinkeby.etherscan.io/address/#{address}
     transaction:     https://rinkeby.etherscan.io/tx/#{txid}
   status:            active
+  protocol:          ERC20
```
Comment (chumaknadya) [inline config/templates/seed/blockchains.yml.erb]: update protocols

File: db/migrate/20210601111215_multi_network_support.rb [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    create_table :blockchain_currencies do |t|
+      t.string :currency_id, foreign_key: true, class: 'Currency', null: false
+      t.string :blockchain_key, foreign_key: true, null: true, class: 'Blockchain'
+      t.string :parent_id, index: true
+      t.decimal :deposit_fee, precision: 32, scale: 16, default: 0, null: false
+      t.decimal :min_deposit_amount, precision: 32, scale: 16, default: 0, null: false
+      t.decimal :min_collection_amount, precision: 32, scale: 16, default: 0, null: false
+      t.decimal :withdraw_fee, precision: 32, scale: 16, default: 0, null: false
+      t.decimal :min_withdraw_amount, precision: 32, scale: 16, default: 0, null: false
+      t.decimal :withdraw_limit_24h, precision: 32, scale: 16, default: 0, null: false
```
Comment (mnaichuk) [inline db/migrate/20210601111215_multi_network_support.rb]: we can drop withdraw limits from currencies

File: docs/peatio/multi_network.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+- BSC
+- ...
+
+Even BTC and ETH are tokenized on Ethereum and others blockchains
+
+## Platform Configurations
+
+### Blockchains
+
+New fields added to blockchain configuration, now you can configure min deposit amount, withdraw fee, min withdraw amount on blockchain level
```
Comment (mnaichuk) [inline docs/peatio/multi_network.md]: in platform currency

File: docs/peatio/multi_network.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+```
+min_deposit_amount = (min deposit amount from blockchain configurations) / (middle price from the platform market CCY/PLATFORM_CURRENCY)
+
+withraw_fee = (withraw fee from blockchain configurations) / (middle price from the platform market CCY/PLATFORM_CURRENCY)
+
+min_withdrawal_amount = (min withdrawal amount from blockchain configurations) / (middle price from the platform market CCY/PLATFORM_CURRENCY)
+
+```
+
+If there is no market pair CCY/PLATFORM_CURRENCY, system will use price from currency configuration
```
Comment (mnaichuk) [inline docs/peatio/multi_network.md]: update this pls

**Review by calj — APPROVED** [review]

**Pull request merged**